### PR TITLE
Fix MKCOL for IE11 as well

### LINKED ISSUE #23067: OC9 and IE 11 - cannot create new folders in WebUI - MKCOL exception recorded
Fresh Install of OC9, when logging in with Internet Explorer v11 - cannot create new folders through WebUI. I can create folders with EDGE, Firefox, Chrome browsers without problems.

_(See - reference to issue #22274 where EDGE browser was fixed)_

Setup in short (sorry short setup, but I believe the trail leads to same issue as referenced issue):
• Browser installed in OS win10 with latest patches. (My IE version 11.103.10586.0)
• Clean install of Owncloud 9.

### Steps to reproduce
1. Fresh install of OC9
2. Win10 and IE11 - log in to OC and try to create a new folder.

### Expected behaviour
New folder should be created.

### Actual behaviour
Error - Could not create folder "new folder"  (see log)

### Server configuration
**Operating system**: Linux

**Web server:** Apache 2.2.31

**Database:** MariaDB 5.5.47-0064

**PHP version:**PHP 5.6.11 (cli)

**ownCloud version:** 9.0.0

#### ownCloud log (data/owncloud.log)

**Error Recorded to log:**

Exception: {"Message":"HTTP\/1.1 415 The request body for the MKCOL request must have an xml Content-Type","Exception":"Sabre\DAV\Exception\UnsupportedMediaType","Code":0,"Trace":"#0 [internal function]: Sabre\DAV\CorePlugin->httpMkcol(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\n#1 \/volume1\/web\/owncloud\/3rdparty\/sabre\/event\/lib\/EventEmitterTrait.php(105): call_user_func_array(Array, Array)\n#2 \/volume1\/web\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(459): Sabre\Event\EventEmitter->emit('method:MKCOL', Array)\n#3 \/volume1\/web\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/Server.php(248): Sabre\DAV\Server->invokeMethod(Object(Sabre\HTTP\Request), Object(Sabre\HTTP\Response))\n#4 \/volume1\/web\/owncloud\/apps\/dav\/appinfo\/v1\/webdav.php(55): Sabre\DAV\Server->exec()\n#5 \/volume1\/web\/owncloud\/remote.php(138): require_once('\/volume1\/web\/ow...')\n#6 {main}","File":"\/volume1\/web\/owncloud\/3rdparty\/sabre\/dav\/lib\/DAV\/CorePlugin.php","Line":563,"User":"xxxxxx"}

### PR DESCRIPTION
**Pull request by rullzer**
Fixes https://github.com/owncloud/core/issues/23067

Using https://github.com/owncloud/core/pull/22274 we have to patch the iedavclient.js as well.

It is the exact same work around. I did not even know we had a special IE davclient.

@ketolajuha please verify

CC: @PVince81 @MorrisJobke @nickvergessen 

@karlitschek we should backport this to stable9.

### CODE REVIEW

File: core/js/files/iedavclient.js [hunk 1]
```diff
@@ -39,7 +39,12 @@
 			for(ii in headers) {
 				xhr.setRequestHeader(ii, headers[ii]);
 			}
-			xhr.send(body);
+
+            if (body === undefined) {
```
Comment (nickvergessen) [inline core/js/files/iedavclient.js]: tabs?

Comment (rullzer) [inline core/js/files/iedavclient.js]: mmm now where did those come from... anway fixed..

**Comment by ketolajuha** [comment]
@rullzer 

I can confirm the proposed fix - after adding same to IE dav client, folders can be again created in stable OC 9.0.0 with IE11.

Thanks!!!!!!

**Comment by nickvergessen** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Tested and works :+1: Also still works in other browsers.

**Comment by rullzer** [comment]
Stable9: https://github.com/owncloud/core/pull/23096

**Comment by karlitschek** [comment]
backport is fine. :+1:

**Pull request merged**